modules/programs/tmux.nix

```diff
@@ -263,58 +300,18 @@ in
         home.packages = [ cfg.package ]
           ++ optional cfg.tmuxinator.enable pkgs.tmuxinator
           ++ optional cfg.tmuxp.enable      pkgs.tmuxp;
-
-        home.file.".tmux.conf".text = tmuxConf;
       }
-
-      (mkIf cfg.sensibleOnTop {
-        home.file.".tmux.conf".text = mkBefore ''
-          # ============================================= #
-          # Start with defaults from the Sensible plugin  #
-          # --------------------------------------------- #
-          run-shell ${pkgs.tmuxPlugins.sensible.rtp}
-          # ============================================= #
-        '';
-      })
-
       (mkIf cfg.secureSocket {
         home.sessionVariables = {
           TMUX_TMPDIR = ''''${XDG_RUNTIME_DIR:-"/run/user/\$(id -u)"}'';
         };
       })
 
-      (mkIf (cfg.plugins != []) {
-        assertions = [(
-          let
-            hasBadPluginName = p: !(hasPrefix "tmuxplugin" (pluginName p));
-            badPlugins = filter hasBadPluginName cfg.plugins;
-          in
-            {
-              assertion = badPlugins == [];
-              message =
-                "Invalid tmux plugin (not prefixed with \"tmuxplugins\"): "
-                + concatMapStringsSep ", " pluginName badPlugins;
-            }
-        )];
-
-        home.file.".tmux.conf".text = mkAfter ''
-          # ============================================= #
-          # Load plugins with Home Manager                #
-          # --------------------------------------------- #
-
-          ${(concatMapStringsSep "\n\n" (p: ''
-              # ${pluginName p}
-              # ---------------------
-              ${p.extraConfig or ""}
-              run-shell ${
-                if types.package.check p
-                then p.rtp
-                else p.plugin.rtp
-              }
-          '') cfg.plugins)}
-          # ============================================= #
-        '';
-      })
+      # config file ~/.tmux.conf
+      # use inverse order, because mkMerge inverts it
```

Comment:
Better use `mkBefore` and `mkAfter` to control order since `mkMerge` does not make any guarantee w.r.t ordering.

Reply:
In the update just pushed, I use explicit ordering.